xcodeproj/internal/compilation_providers.bzl

```diff
@@ -133,37 +148,61 @@ def _merge_compilation_providers(
         `compilation_providers.collect`.
     """
     merged_cc_info = cc_common.merge_cc_infos(
-        direct_cc_infos = [cc_info] if cc_info else [],
+        direct_cc_infos = [],
         cc_infos = [
             providers.cc_info
             for _, providers in transitive_compilation_providers
             if providers.cc_info
         ],
     )
 
+    framework_files = memory_efficient_depset(
+        transitive = [
+            providers._propagated_framework_files
+            for _, providers in transitive_compilation_providers
+        ],
+        order = "topological",
+    )
+
+    if apple_dynamic_framework_info:
+        propagated_framework_files = memory_efficient_depset(
+            transitive = [
+                apple_dynamic_framework_info.framework_files,
+                framework_files,
+            ],
+            order = "topological",
+        )
+    else:
+        propagated_framework_files = framework_files
+
```

Comment:
I'm not a fan of this. It all works, but it feels to heavy handed. I'm looking to see if there is a better way.

Reply:
Seems BwX framework detection is broken with the new linking scheme. We get symlinked inputs in `link.params`:

```
bazel-out/applebin_ios-ios_sim_arm64-dbg-ST-6c3cd4e1007a/bin/_solib_ios_sim_arm64/_U_S_SLib_CLibFramework.iOS___ULib/LibFramework.iOS_lipobin.dylib
bazel-out/ios-sim_arm64-min15.0-applebin_ios-ios_sim_arm64-dbg-ST-fda82787d112/bin/_solib_ios_sim_arm64/_U@_Umain~non_Umodule_Udeps~com_Ugithub_Ukrzyzanowskim_Ucryptoswift_S_S_CCryptoSwift___UCryptoSwift.xcframework_Sios-arm64_Ui386_Ux86_U64-simulator_SCryptoSwift.framework/CryptoSwift
```

This is actually better for BwB mode (as long as we can make sure these symlinks and the targets they link to are downloaded correctly with BwtB), because we can do away with `$PREVIEW_FRAMEWORK_PATHS`. For BwX mode though, I'm not sure I can find a way to map these symlinks to the dylib that Xcode produces. Besides for breaking SwiftUI Previews, it breaks linking all together. So I either have to fix it or BwX mode dies.

Reply:
Well, `$PREVIEW_FRAMEWORK_PATHS` is still needed, because of course the `@rpath`s and `LC_ID_DYLIB`s are the same (in order to support bundling).

Reply:
Disabling BwX mode for now.